### ISSUE TITLE
Screencast SVG incurs a rendering burden on the browser

### ISSUE THREAD

**Issue by ivanreese** [lycheeverse/lychee#1451]
When I load the main repo page in Safari, the screencast SVG renders in a really slow and glitchy way (image below). What's worse, it brings scrolling performance to a crawl (like 2 fps) even when the SVG isn't on screen. Safari struggles to keep up as I scroll down the README, showing large blank sections.

I realize the SVG is smaller than a GIF of the same content — I read the PR discussion about it — but the few dozen KB saved has instead caused a pretty significant amount of layout/paint/compositing work to be done in the browser. (Eg: testing in Chrome, it uses about 50% CPU when the SVG is showing full-window scrolling text). Not sure that's a worthwhile tradeoff, considering the baseline page weight of Github is already around 1MB over the wire, so you're not saving much data, percentage-wise.

I'm on an M1 Pro MBP, so I worry this experience would be even worse for people on less capable hardware.

Anyway, do with this info what you will — I just figure it was worth alerting you to the issue.

![CleanShot 2024-06-17 at 19 33 53](https://github.com/lycheeverse/lychee/assets/2061924/cae66870-3d74-4b41-96aa-22557d075f0a)

**Comment by mre** [comment]
Hey @ivanreese,

Thanks for the heads-up.

I also have a MacBook M1 (Max, not Pro, but I doubt that matters). The performance is butter-smooth on Firefox and Chrome. I just tried Safari, and I noticed the framerate drops, but I didn't see any artifacts. That's very weird.

Can others comment on this? If it continues to be a problem, I wouldn't mind switching to a video or a GIF.

By the way, the main reason for using SVG wasn't size for me but the fact that glyphs are super sharp at any resolution. I like the crisp look.

**Comment by mre** [comment]
@ivanreese, can you still reproduce this? Do you still see artifacts?
Also, if someone else can double-check and report back, that would be great.

**Comment by ivanreese** [comment]
I no longer see artifacts, but it still tanks the framerate, especially when I try to scroll, even if the SVG is offscreen. Looking at the Timeline view in dev tools, I'm seeing the CPU hover around 80-90%.

**Comment by mre** [comment]
Thanks for the feedback. I'll keep this open then. We need to look into how to fix this. Perhaps we can reduce the frame rate of the animation.

**Comment by mre** [comment]
Just tested the status again on two laptops and it works completely fine on all browsers, including Safari. Also tested on my phone and it works okay. Could it be a browser extension maybe? I'm inclined to close this for now, but please reopen in case it's still an issue. Thanks for reporting.

**Comment by ivyreese** [comment]
Yeah, it still affects me. Not running any extensions, I'm on Safari 26.3. This is literally the only page where I've seen this issue.

If nobody else has complained, it probably doesn't matter. I'm sure it only affects a small number of people. No need to re-open, and thanks for following up.
